### PR TITLE
Use vector implementation for `calc_alan_entries()`

### PR DESCRIPTION
**Pull request by smokestacklightnin**
### :pencil: Description

**Type:** :rocket: `feature` 

Use vectorized implementations for the body of `calc_alan_entries()`.

### :vertical_traffic_light: Testing

How did you test these changes?

- [X] Testing pipeline

### :ballot_box_with_check: Checklist

- [x] I requested two reviewers for this pull request
- [ ] I updated the documentation according to my changes

### CODE REVIEW

File: stardis/radiation_field/opacities/opacities_solvers/base.py [hunk 1]
```diff
@@ -516,17 +518,17 @@ def calc_alan_entries(
     float
         Line opacity.
     """
+    delta_nu = np.abs(delta_nus)
+    doppler_width = doppler_widths_at_depth_point
+    gamma = gammas_at_depth_point
```
Comment (jvshields) [inline stardis/radiation_field/opacities/opacities_solvers/base.py:523]: I don't understand why each of these are redefined, or the need for the absolute value if voigt profiles are symmetric. Can the three function inputs be fed directly into the voigt profile function? And if so, why do we want to allocate a bit of extra memory here?

Comment (smokestacklightnin) [reply stardis/radiation_field/opacities/opacities_solvers/base.py:523]: In the case of `delta_nu`, the absolute value was there before I changed the code, so I left it alone. This is a question about the physical meaning of the formula.

In the case of `doppler_width` and `gamma`, no new memory is allocated. The names are just bound to the data pointed to by `doppler_widths_at_depth_point` and `gammas_at_depth_point`, respectively. See the demo:
image

Comment (jvshields) [reply stardis/radiation_field/opacities/opacities_solvers/base.py:523]: Neat. I learned something new about python today. Though I still think simply using the inputs where appropriate to save line space might be the way to go for brevity. 

As for the absolute value, gotcha and I suppose a conversation I should take up with whoever wrote these lines originally.

Comment (smokestacklightnin) [reply stardis/radiation_field/opacities/opacities_solvers/base.py:523]: Interestingly, "[Python accesses local variables much more efficiently than global variables.](https://wiki.python.org/moin/PythonSpeed/PerformanceTips#Local_Variables)."

I don't think that makes a difference here, so I agree with you in this particular case.

**Pull request closed without merging**